WordPress/openverse
# Fix unit test warnings

## Fixes



Related to #5171 by @obulat

## Description



This PR introduces multiple improvements to the unit tests to eliminate warnings and produce cleaner output.

- Removed the custom `render-suspended` function we used to override `nuxt/test-utils`. Since we now consistently use script setup instead of defineComponent, the default render-suspended function works seamlessly. 

- Updated component props. Several components' props have changed, so I updated them in tests.

- Added ESLint Rule for props with defaults: `"vue/no-required-prop-with-default"` ESLint rule marks a prop optional if this prop has a default value. This helped fix a couple of warnings and components ( for example,`VHomeLink`, `VLogoloader`).

- Updated issue Link: Corrected the issue link to point to the tracker for unit test fixes following the migration to Nuxt 3. 
- Addressed prop fall-through in complex components: 
When a component is complex, such as a `VTag` that `:is="VButton"` that `:is="VLink"` (3 levels of nestedness), the prop does not always fall through, at least in the unit tests, so I added manual binding of the `href` to `VButton` if the `component` in `VButton` is a `VLink`.

## Testing Instructions

Run the unit tests using `ov just frontend/run test:unit`.
You should see much fewer warnings.

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
